src/Nethermind.Int256/UInt256.cs

```diff
@@ -933,7 +933,7 @@ private static bool SubtractImpl(in UInt256 a, in UInt256 b, out UInt256 res)
 
         public void Subtract(in UInt256 b, out UInt256 res) => Subtract(this, b, out res);
 
-        public static void SubtractMod(in UInt256 a, in UInt256 b, in UInt256 m, out UInt256 res)
```

Comment:
Any reason to remove `in` here?

Reply:
Yes, I made it intentionaly.  
This will create a copy of `a`.
So when you use it like
`UInt256.SubtractMod(a, in b, out a) `
and inside the method you reach 
`if (SubtractUnderflow(a, b, out res))`
this line will not override `a`  (as `a` and `res` are the same) (and we need original value of `a` later)

I could just made a copy of `a` inside the method instead, but why do we need to pass it as reference and then just copy it?

Same situation you can find in
`static void Exp(in UInt256 b, in UInt256 e, out UInt256 result)`
where I removed a line `UInt256 bs = b;` and change signature to
`public static void Exp(UInt256 b, in UInt256 e, out UInt256 result)`

Same in ExpMod

Reply:
|                         Method | EnvironmentVariables |      Mean |     Error |    StdDev | Ratio |
|------------------------------- |--------------------- |----------:|----------:|----------:|------:|
|            SubtractMod_UInt256 |                Empty |  5.878 ns | 0.0035 ns | 0.0029 ns |  1.00 |
|  SubtractMod_UInt256_withoutIn |                Empty |  5.741 ns | 0.0039 ns | 0.0032 ns |  0.98 |

Reply:
And if I will change
```
public static void Multiply(in Int256 a, in Int256 b, out Int256 res)
{
    Int256 av = a, bv = b;
    if (a.Sign < 0)
    {
        a.Neg(out av);
    }
    if (b.Sign < 0)
    {
        b.Neg(out bv);
    }
    UInt256.Multiply(av._value, bv._value, out UInt256 ures);
    int aSign = a.Sign;
    int bSign = b.Sign;
    res = new Int256(ures);
...
```
to
```
public static void MultiplyWithoutIn(Int256 a, Int256 b, out Int256 res)
{
    int aSign = a.Sign;
    int bSign = b.Sign;

    if (aSign < 0) a.Neg(out a);
    if (bSign < 0) b.Neg(out b);
    UInt256.Multiply(a._value, b._value, out UInt256 ures);

    res = new Int256(ures);
...
```
(remove `in` and `Int256 av = a, bv = b;`)
 this will give improvement too

Apple M1 Pro, .NET 7.0 : .NET 7.0.5 (7.0.523.17405), Arm64 RyuJIT AdvSIMD

|                    Method |       EnvironmentVariables |      Mean | Ratio |
|-------------------------- |--------------------------- |----------:|------:|
|           Multiply_Int256 |                      Empty |  9.861 ns |  1.00 |
| Multiply_Int256_withoutIn |                      Empty |  9.463 ns |  0.96 |
|           Multiply_Int256 | DOTNET_EnableHWIntrinsic=0 | 19.376 ns |  1.96 |
| Multiply_Int256_withoutIn | DOTNET_EnableHWIntrinsic=0 | 18.132 ns |  1.84 |